File: domain/src/main/java/com/michaelfotiadis/samplearchitecture/domain/listener/Canceller.java
package com.michaelfotiadis.samplearchitecture.domain.listener;

public abstract class Canceller {

    private Cancellable cancellable;

    public void register(Cancellable cancellable) {
        this.cancellable = cancellable;
    }

    public void unRegister() {
        this.cancellable = null;
    }

    protected void cancel() {
        if (cancellable != null) {
            cancellable.cancel();
        }
    }

}

File: domain/src/main/java/com/michaelfotiadis/samplearchitecture/domain/listener/Canceller.kt
package com.michaelfotiadis.samplearchitecture.domain.listener

abstract class Canceller {

    private var cancellable: Cancellable? = null

    fun register(cancellable: Cancellable) {
        this.cancellable = cancellable
    }

    fun unRegister() {
        this.cancellable = null
    }

    protected fun cancel() {
        if (cancellable != null) {
            cancellable!!.cancel()
        }
    }

}
